File: 24_C#Fundamentals/_05_Lists/_02_Exercise/_06_CardsGame/_06_CardsGame.cs
using System;
using System.Xml.Linq;
using System.Linq;

public class _06_CardsGame
{
    public static void Main(string[] args)
    {
        // Read input values
        List<int> cardsList1 = Console.ReadLine().Split().Select(int.Parse).ToList();
        List<int> cardsList2 = Console.ReadLine().Split().Select(int.Parse).ToList();

        // Play card game and determine who wins
        while (cardsList1.Count > 0 && cardsList2.Count > 0)
        {
            if (cardsList1[0].Equals(cardsList2[0]))
            {
                cardsList1.RemoveAt(0);
                cardsList2.RemoveAt(0);
            }
            else if (cardsList1[0] > cardsList2[0])
            {
                cardsList1.Add(cardsList1[0]);
                cardsList1.Add(cardsList2[0]);

                cardsList1.RemoveAt(0);
                cardsList2.RemoveAt(0);
            }
            else if (cardsList1[0] < cardsList2[0])
            {
                cardsList2.Add(cardsList2[0]);
                cardsList2.Add(cardsList1[0]);

                cardsList1.RemoveAt(0);
                cardsList2.RemoveAt(0);
            }
        }

        int sum = 0;

        // Print result
        if (cardsList1.Count == 0)
        {
            foreach (int integer in cardsList2)
            {
                sum += integer;
            }

            Console.WriteLine($"Second player wins! Sum: {sum}");
        }
        else
        {
            foreach (int integer in cardsList1)
            {
                sum += integer;
            }

            Console.WriteLine($"First player wins! Sum: {sum}");
        }
    }
}

File: 02_ProgrammingFundamentals/_05_Lists/_02_Exercise/_06_CardsGame.java
package _02_ProgrammingFundamentals._05_Lists._02_Exercise;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _06_CardsGame {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<Integer> cardsList1 = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
        List<Integer> cardsList2 = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());

        // Play card game and determine who wins
        while (cardsList1.size() > 0 && cardsList2.size() > 0) {
            if (cardsList1.get(0).equals(cardsList2.get(0))) {
                cardsList1.remove(0);
                cardsList2.remove(0);
            } else if (cardsList1.get(0) > cardsList2.get(0)) {
                cardsList1.add(cardsList1.get(0));
                cardsList1.add(cardsList2.get(0));

                cardsList1.remove(0);
                cardsList2.remove(0);
            } else if (cardsList1.get(0) < cardsList2.get(0)) {
                cardsList2.add(cardsList2.get(0));
                cardsList2.add(cardsList1.get(0));

                cardsList1.remove(0);
                cardsList2.remove(0);
            }
        }

        int sum = 0;

        // Print result
        if (cardsList1.isEmpty()) {
            for (Integer integer : cardsList2) {
                sum += integer;
            }

            System.out.printf("Second player wins! Sum: %d%n", sum);
        } else {
            for (Integer integer : cardsList1) {
                sum += integer;
            }

            System.out.printf("First player wins! Sum: %d%n", sum);
        }

        scan.close();
    }
}
